src/shared/GitHub/GitHubHostProvider.cs

```diff
@@ -316,6 +316,11 @@ public static bool IsGitHubDotCom(string targetUrl)
 
         public static bool IsGitHubDotCom(Uri targetUri)
         {
+            if (targetUri is null)
+            {
+                throw new ArgumentNullException(nameof(targetUri));
+            }
```

Comment:
We have a set of arg validation helpers that would be better to use here:

```suggestion
            EnsureArgument.NotNull(targetUri, nameof(targetUri));
```

..or for a URI, might even be better to validate for an absolute URI:

```suggestion
            EnsureArgument.AbsoluteUri(targetUri, nameof(targetUri));
```

(this also checks for `null` too)